psvpfsparser/PfsFile.cpp

```diff
@@ -125,11 +129,20 @@ int PfsFile::decrypt_icv_file(boost::filesystem::path destination_root) const
       }
       else
       {
-         outputStream.write((char*)buffer.data(), read_size);
+         bytes_left = bytes_left - read_size;
+         sector_base = sector_base + num_sectors;
+         if (read_size <= real_bytes_left)
+         {
+            outputStream.write((char*)buffer.data(), read_size);
+            real_bytes_left -= read_size;
+         }
+         else
+         {
+            outputStream.write((char*)buffer.data(), real_bytes_left);
+            real_bytes_left = 0;
+            break;
```

Comment:
I am not sure if this is a good idea to break here.
If real_bytes_left is 0 and for cycle with blocks did not exit - it means something went wrong.
It is better to throw and error (return -1 and write an error to m_output) before trying to do a decryption if real_bytes_left != 0

Reply:
changed

https://github.com/motoharu-gosuto/psvpfstools/blob/94471aefbc4e991bc2981f1489237df9c7ff44f2/psvpfsparser/PfsFile.cpp#L132-L146

https://github.com/motoharu-gosuto/psvpfstools/blob/94471aefbc4e991bc2981f1489237df9c7ff44f2/psvpfsparser/PfsFile.cpp#L158-L162

Reply:
perfect